### PR TITLE
feat: Add plugin-kit package

### LINKED ISSUE #97: Change Request: New "helper" package for plugins
### Which packages would you like to change?

- [ ] `@eslint/compat`
- [ ] `@eslint/config-array`
- [ ] `@eslint/core`
- [ ] `@eslint/migrate-config`
- [ ] `@eslint/object-schema`

### What problem do you want to solve?

Now that I've worked on a couple of language plugins, it's clear that there's some duplication of logic that each language could have. For instance:

- A basic `SourceCode` implementation
- Classes for `VisitTraversalStep` and `CallTraversalStep`
- Configuration comment parsing

### What do you think is the correct solution?

Rather than forcing everyone to write their own versions of this logic, it would be helpful to create a package that contains common utilities for language development.

Maybe `@eslint/language`? Or `@eslint/plugin` or `@eslint/plugin-kit` (to disambiguate with ESLint plugins and make it a more generic plugin helper package rather than a language plugin package?)

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

_No response_

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Created the new `plugin-kit` package

#### What changes did you make? (Give an overview)

- Created `plugin-kit` directory
- Created `ConfigCommentParser`, `VisitNodeStep`, and `CallMethodStep` classes
- Added tests for each class
- Updated release-please config
- Updated release-please CI job
- Updated issue templates

#### Related Issues

fixes #97

#### Is there anything you'd like reviewers to focus on?

Just starting with the basics for this package. These are some commonalities I've found across the different languages. There will likely be more.

I slightly changed the interface of `ConfigCommentParser` to be more generic as the one in `eslint/eslint` was very specific to the way Espree represents comments. I tested this `ConfigCommentParser` in ESLint directly to ensure the new interface works correctly.

### CODE REVIEW

**Review by harish-sethuraman — APPROVED** [review]
LGTM. Went through the files related to actions/readme/build. Will wait for others to review files in source and test folders

File: packages/plugin-kit/src/config-comment-parser.js
Comment (nzakas) [reply packages/plugin-kit/src/config-comment-parser.js:142]: Yes that's correct.

File: packages/plugin-kit/src/config-comment-parser.js
Comment (nzakas) [reply packages/plugin-kit/src/config-comment-parser.js:6]: Making it static class means we'll never be able to pass parameters in, and I think there's a high likelihood that we will do so in the future.

File: packages/plugin-kit/jsr.json
Comment (nzakas) [reply packages/plugin-kit/jsr.json:10]: I'm following the convention we have in the other packages. If we want to change that convention, let's do that for all packages in a separate PR.

**Review by aladdin-add — APPROVED** [review]
LGTM. Would like another review before merging.

File: packages/core/src/types.ts
Comment (mdjermanovic) [inline packages/core/src/types.ts:405]: Why this change, isn't this expected to be an object (AST node)?

File: release-please-config.json
Comment (mdjermanovic) [inline release-please-config.json:51]:
```suggestion
```

I think this isn't necessary, and we'd need to remove it after the first release so seems better not to add it.

Comment (mdjermanovic) [reply release-please-config.json:51]: Actually it does seem necessary if we want to start with 0.x, I see now that new packages were released as 1.0.0.

File: packages/plugin-kit/README.md
Comment (mdjermanovic) [inline packages/plugin-kit/README.md:72]:
```suggestion
console.log(Object.entries(strings)); // [["foo", "off"], ["bar", null]]
```

File: packages/plugin-kit/README.md
Comment (mdjermanovic) [inline packages/plugin-kit/README.md:78]:
```suggestion
console.log(Object.entries(jsonLike.config)); // [["semi", ["error", "never"]], ["prefer-const", "warn"]]
```

File: packages/core/src/types.ts
Comment (nzakas) [reply packages/core/src/types.ts:405]: `unknown` forces consumers to specify the type of the object before working with it, which ensures that it's properly typed.

`object` will just cause squiggles every time you access a property or method that isn't on `Object.prototype`.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
